### PR TITLE
Feat: Add all_burst_number

### PR DESCRIPTION
**Pull request by arnaudon**
## Description

See docstring for details, WIP for work with @PolinaL 

I'll add test/doc/changelog once stable 

## Checklist:
- [ ] Unit tests are added to cover the changes (skip if not applicable).
- [ ] The changes are mentioned in the documentation (skip if not applicable).
- [ ] CHANGELOG file is updated (skip if not applicable).

### CODE REVIEW

File: efel/pyfeatures/pyfeatures.py [hunk 1]
```diff
@@ -140,6 +141,45 @@ def strict_burst_number() -> np.ndarray:
     return np.array([burst_mean_freq.size])
 
 
+def all_burst_number():
```
Comment (anilbey) [inline efel/pyfeatures/pyfeatures.py]: Wouldn't it be more descriptive to call it something like  `count_burst_events_by_isi_clustering`?

File: efel/pyfeatures/pyfeatures.py [hunk 2]
```diff
@@ -140,6 +141,45 @@ def strict_burst_number() -> np.ndarray:
...
+    stim_start = _get_cpp_data("stim_start")
+    stim_end = _get_cpp_data("stim_end")
+    peak_times = get_cpp_feature("peak_time")
+    if peak_times is None:
+        return np.array([0])
+
+    peak_times = peak_times[(peak_times > stim_start) & (peak_times < stim_end)]
+    isis = np.diff(peak_times)
+
+    from sklearn.cluster import KMeans
```
Comment (anilbey) [inline efel/pyfeatures/pyfeatures.py:161]: Could you put it at the top of the file and then add `scikit-learn>=X.Y.Z` to `setup.py`, where X.Y.Z is the version you tested?

File: efel/pyfeatures/pyfeatures.py [hunk 3]
```diff
@@ -140,6 +141,46 @@ def strict_burst_number() -> np.ndarray:
     return np.array([burst_mean_freq.size])
 
 
+def all_burst_number(raise_warnings: bool = False) -> np.ndarray:
```
Comment (anilbey) [inline efel/pyfeatures/pyfeatures.py:144]: Wouldn't it be more descriptive to name it something like `count_burst_events_by_isi_clustering`?

**Review by anilbey — CHANGES_REQUESTED** [review]
Thanks @arnaudon  looks very good. I added a few remarks.

Could you also add a test to it?

In the test could you check when number of isis is <= 1?

Also, will it work when the number of isis is equal to 2?

**Comment by arnaudon** [comment]
I'm closing this, we can use https://github.com/BlueBrain/eFEL/pull/391

**Pull request closed without merging**